src/Uno.UI/UI/Xaml/ResourceDictionary.cs

```diff
@@ -249,6 +249,7 @@ private void Set(in ResourceKey resourceKey, object value, bool throwIfPresent)
 			{
 				_hasUnmaterializedItems = true;
 				_values[resourceKey] = new LazyInitializer(ResourceResolver.CurrentScope, resourceInitializer);
+				ResourceDictionaryValueChange?.Invoke(this, EventArgs.Empty);
```

Comment:
Did you check how often this is called? It seems like XamlFileGenerator is only using `WeakResourceInitializer`, and `ResourceInitializer` is only added by `ResourceResolver.RegisterResourceDictionaryBySource()`. So maybe it's not called often.